### PR TITLE
Feat: login, logout func implemented

### PR DESCRIPTION
**Pull request by Logan1x**
- login, logout func implemented
- created content for auth

### CODE REVIEW

File: src/components/nav.js [hunk 1]
```diff
@@ -26,9 +31,15 @@ export default function nav() {
         </svg>
       </div>
       <div className="nav-pills">
-        <a href="/auth/login.html" className=" btn-filled">
```
Comment (shrey27) [inline src/components/nav.js:29]: use Link tags here as well

File: src/pages/login.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+export default function Login() {
+  const [email, setEmail] = useState("khu@gmail.com");
+  const [password, setPassword] = useState("khush");
+
+  const { loginHandler } = useAuth();
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    loginHandler(email, password);
```
Comment (shrey27) [inline src/pages/login.jsx:14]: handle empty cases as well and maybe show some error messages

**Review by shrey27 — APPROVED** [review]
Overall code looks good

PR is good to merge

File: src/components/nav.js (hunk 1 above)
Comment (Logan1x) [reply src/components/nav.js:29]: Link tag needs mandatory `to` attribute, and for this, I didn't need that. so when I was using it without `to`, there were errors.

File: src/contexts/authContext.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import { useNavigate } from "react-router-dom";
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const navigate = useNavigate();
+
+  const localStorageToken = JSON.parse(localStorage.getItem("loginItems"));
+  const [token, setToken] = useState(localStorageToken?.token);
+  const [currUser, setCurrUser] = useState(localStorageToken?.user);
```
Comment (Rohanmond) [inline src/contexts/authContext.js:12]: liked the way of using optional chaining

File: src/index.js [hunk 4]
```diff
@@ -12,7 +13,9 @@ ReactDOM.render(
   <React.StrictMode>
     <BrowserRouter>
       <DataProvider>
-        <App />
+        <AuthProvider>
```
Comment (Rohanmond) [inline src/index.js:16]: AuthProvider can put as a parent of DataProvider

**Review by Rohanmond — APPROVED** [review]

**Comment by Logan1x** [comment]
Thank you both, Merging the PR :)

File: src/index.js (hunk 4 above)
Comment (Logan1x) [reply src/index.js:16]: done, will be in next commit.

**Pull request merged**